internal/config/config.go

```diff
@@ -145,8 +145,8 @@ type AzureConfig struct {
 	//   Authorize spawned VMs to access Azure API.
 	UserAssignedIdentity string `yaml:"userAssignedIdentity" validate:"required"`
 	// description: |
-	//    Application client ID of the Active Directory app registration.
-	AppClientID string `yaml:"appClientID,omitempty" validate:"omitempty,uuid"`
+	//  DEPRECATED:  Application client ID of the Active Directory app registration.
+	AppClientID string `yaml:"appClientID,omitempty" validate:"omitempty,deprecated,deprecatedAppClientID"`
```

Comment:
We need to remove the config key from the struct completely. Otherwise, each newly generated config also includes the key. 

I think a way to provide an extra error message for the field being set is to iterate over the ValidationErrors [1] and check the field name [2]. Keys that are not defined in the Config struct produce an error already. Then you can wrap the validation error or sth similar.

[1] https://github.com/edgelesssys/constellation/blob/main/internal/config/config.go#L731
[2] https://pkg.go.dev/github.com/go-playground/validator/v10#FieldError